templates/nginx.centos.repo.j2

```diff
@@ -0,0 +1,13 @@
+{% if nginx_el_install_mainline %}
+[nginx]
+name=nginx repo mainline
+baseurl=http://nginx.org/packages/mainline/centos/$releasever/$basearch/
```

Comment:
You can change the `centos` here to: `{{ ansible_distribution | lower }}`

I don't have a RHEL server around to verify it outputs "RHEL" though... can someone else check?

Worst case you could add a little conditional logic to print `centos` or `rhel`. I'd rather have one file.